flow/connectors/postgres/cdc.go

```diff
@@ -297,17 +297,19 @@ func (p *PostgresCDCSource) consumeStream(
 			}
 		}
 
-		if (cdcRecordsStorage.Len() >= int(req.MaxBatchSize)) && !p.commitLock {
-			return nil
-		}
+		if !p.commitLock {
+			if cdcRecordsStorage.Len() >= int(req.MaxBatchSize) {
```

Comment:
Cleaned up this to potentially add [activity.GetWorkerStopChannel](https://pkg.go.dev/go.temporal.io/sdk/activity#GetWorkerStopChannel) handling, but in this ended up not helping for e2e testing & so integrating that can be left as something for the future